gnomad/utils/intervals.py

```diff
@@ -112,3 +112,53 @@ def _add_padding(
         return [_add_padding(i) for i in intervals]
     else:
         return _add_padding(intervals)
+
+
+def explode_intervals_to_loci(
+    obj: Union[hl.Table, hl.MatrixTable],
+    interval_field: str = "interval",
+    keep_intervals: bool = False,
+) -> Union[hl.Table, hl.MatrixTable]:
+    """
+    Expand intervals to loci.
+
+    :param obj: Hail Table or MatrixTable with an interval field.
+    :param interval_field: Name of the interval field. Default is 'interval'.
+    :param keep_intervals: If True, keep the original interval as a column in output.
+    :return: Hail Table or MatrixTable with interval exploded to loci.
+    """
+    is_matrix = isinstance(obj, hl.MatrixTable)
+    ht = obj.rows() if is_matrix else obj
+
+    interval = ht[interval_field]
+    includes_start = interval.includes_start.take(1)[0]
+    includes_end = interval.includes_end.take(1)[0]
+
+    interval_start = (
+        interval.start.position if includes_start else interval.start.position + 1
+    )
+    interval_end = interval.end.position + 1 if includes_end else interval.end.position
```

Comment:
```suggestion
    interval_start_expr = hl.if_else(interval_expr.includes_start, interval_expr.start.position, interval_expr.start.position + 1)
    interval_end_expr = hl.if_else(interval_expr.includes_end, interval_expr.end.position + 1, interval_expr.end.position)
```
this avoids the conversion from Hail to python with `take` (though it might not be properly formatted)